src/services/codecovEventMetrics/useStoreCodecovEventMetric.tsx

```diff
@@ -21,14 +21,25 @@ const mutationQuery = `
     }
   }
 `
+// Maintain parity with list of allowed metrics here
+// https://github.com/codecov/shared/blob/main/shared/django_apps/codecov_metrics/service/codecov_metrics.py
+export const EVENT_METRICS = {
+  VISITED_PAGE: 'VISITED_PAGE',
+  CLICKED_BUTTON: 'CLICKED_BUTTON',
+  COPIED_TEXT: 'COPIED_TEXT',
+  COMPLETED_UPLOAD: 'COMPLETED_UPLOAD',
+  INSTALLED_APP: 'INSTALLED_APP',
+} as const
```

Comment:
It's a tradeoff, in Rohit's approach the optimization is being traded off for a better auto-complete experience in the IDE.

Reply:
I prefer to use a single object as a source of truth for all metrics as it is more structured that way. IMO the benefits from readability and accessibility of the structure outweigh any minification  optimizations on a few lines, given just a small number of constants.

Happy to make the change if you think there are any practical performance concerns from not doing this.

Reply:
Giving this a bit more thought, I think we should take a slightly different approach then both the current implementation and my original suggestion, my original reasoning isn't all that great. 

I think we should remove the object entirely because it's introduces a lot of duplicate code both the key and value are always the same, and using the object that way also introduces the possibility that only one side of the key or value is changed but not the other, making this code brittle.

The approach I think would solve all the concerns and be very ergonomic is to create a union type and just use string literals in place. Right now the typeahead will only work after you imported the object, whereas with a type union it works instantly after you open a leading quote. 

Here's an example of what the union type would look like:

```typescript
type EventMetric =  'VISITED_PAGE' | 'CLICKED_BUTTON' | 'COPIED_TEXT' | 'COMPLETED_UPLOAD' | 'INSTALLED_APP' 
```

Reply:
Similarly, we can utilize TS enums here if we want something similarly ergonomic.
https://www.typescriptlang.org/docs/handbook/enums.html#enums-at-compile-time

This example is somewhat of a middle ground between both of your approaches, just a little easier for me to read personally without all the '|'  as we add more metrics but also not as verbose settings the keys and values

```
enum EVENT_METRICS {
  VISITED_PAGE,
  CLICKED_BUTTON,
  COPIED_TEXT,
  COMPLETED_UPLOAD,
  INSTALLED_APP
}

type EventMetrics = keyof typeof EVENT_METRICS

const mymetric:EventMetrics = 'lol' // This errors because it's looking for the union string type
```

Reply:
Can you add some comments around the enum, to inform people to not actually ever use the enum directly. There are quite a few gotcha's with enums, that we want to avoid at all costs so we don't want people to accidentally use it and cause larger issues.

You can read more about the issues with enums, and I personally think we should avoid them, "[Why I Don't Like Enums]"(https://www.totaltypescript.com/why-i-dont-like-typescript-enums)

Reply:
These are good points, but I think for a change a small as this, how we choose to implement it will have a negligible effect on performance or any other meaningful metric. 

I will merge this change as is later today unless we think there is a strong practical reason to go with one approach over the other.

Reply:
Sounds good